src/makeCalculator.js

```diff
@@ -38,6 +38,32 @@
  */
 function makeCalculator() {
   // write code here
+  return {
+    result: 0,
+
+    add(a, b) {
+      return a + b;
+    },
+    subtract(a, b) {
+      return a - b;
+    },
+    multiply(a, b) {
+      return a * b;
+    },
+    divide(a, b) {
+      return a / b;
+    },
+    reset() {
+      this.result = 0;
+
+      return this;
+    },
+    operate(callback, number) {
+      this.result = callback(this.result, number);
```

Comment:
The 'operate' method should check that 'callback' is a function before calling it, to prevent potential errors.